src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -146,21 +147,30 @@ public void updatePerson(ReadOnlyPerson target, ReadOnlyPerson editedPerson)
 
     //@@author freesoup
     @Override
-    public void removeTag(Tag tag) throws PersonNotFoundException, DuplicatePersonException {
+    public void removeTag(Tag tag) throws PersonNotFoundException, DuplicatePersonException,
+            NoSuchTagException {
+        if (!addressBook.getTagList().contains(tag)) {
+            throw new NoSuchTagException();
+        }
         ObservableList<ReadOnlyPerson> list = addressBook.getPersonList();
-
         for (int i = 0; i < list.size(); i++) {
             ReadOnlyPerson person = list.get(i);
             removeTagFromPerson(tag, person);
         }
+        addressBook.removeTagFromUniqueList(tag);
         indicateAddressBookChanged();
     }
 
     @Override
-    public void removeTag(Index index, Tag tag) throws PersonNotFoundException, DuplicatePersonException {
+    public void removeTag(Index index, Tag tag) throws PersonNotFoundException, DuplicatePersonException,
```

Comment:
`PersonNotFoundException` is understandable, but why `DuplicatePersonException` is declared to in `throws` throws me off.